pkg/util/env/env.go

```diff
@@ -40,22 +41,23 @@ const (
 // GetNodeName returns the node's name used in Kubernetes, based on the priority:
 // - Environment variable NODE_NAME, which should be set by Downward API
 // - OS's hostname
-func GetNodeName() (string, error) {
-	nodeName := os.Getenv(NodeNameEnvKey)
+func GetNodeName() (nodeName string, err error) {
+	defer func() {
+		if runtime.IsWindowsPlatform() {
+			nodeName = strings.ToLower(nodeName)
+		}
+	}()
+	nodeName = os.Getenv(NodeNameEnvKey)
 	if nodeName != "" {
-		return nodeName, nil
+		return
 	}
 	klog.Infof("Environment variable %s not found, using hostname instead", NodeNameEnvKey)
-	var err error
 	nodeName, err = os.Hostname()
 	if err != nil {
 		klog.Errorf("Failed to get local hostname: %v", err)
-		return "", err
-	}
-	if runtime.GOOS == "windows" {
-		return strings.ToLower(nodeName), nil
+		return
 	}
-	return nodeName, nil
+	return
```

Comment:
Original node not set the value of envrionment variable NODE_NAME to lower case.

Reply:
Where does the environment variable NODE_NAME come from in this case?

Reply:
It is read from the environment. For Windows, we need to think of the case Agent is running as a process, then this environment variable is set by user but not kubelet. Since the existing code has logic to ensure the Windows hostname is transformed to lowercase, I think it should be also suitable for the user added environment variable value. 

Besides, if your concern is named result, I think I could modify it. But I want to learn your point about if we need to ensure the environment variable value is lowercase on Windows or not.

Reply:
I don't think Kubernetes allows to use upper case for name field. At least, kubelet uses lowercase for node name. Windows host is not sensitive on the upper case or lowercase, so user is possible to use uppercase in the environment variable. But if antrea uses the value to query Node directly, it should fail. That's why I think we should also transform the value of environment variable to lowercase in this change.

Reply:
Yes, I understand Kubernetes doesn't allow upper case for name fields. I meant should we limit the case strictly like Kubernetes does since we cannot support upper case, instead of allowing user to set it but converting it implicitly. When we ask user to set this environment variable, there should be a doc about the supported characters and case, right? But this is minor anyway.